.github/workflows/build_and_test.yml

```diff
@@ -297,6 +298,80 @@ jobs:
           fail_ci_if_error: false
           verbose: true
 
+  macos-m1:
+    runs-on: ${{ matrix.os-version }}
+    strategy:
+      matrix:
+        os-version: [macos-14]
```

Comment:
so there will no MacOS x86 runner on macos-14 ?

Reply:
yes, according to github official blog, macos-14 is m1 only
https://github.blog/changelog/2024-01-30-github-actions-introducing-the-new-m1-macos-runner-available-to-open-source/

Reply:
Really strange because they have runner images for macos-14 nad macos-14-arm64
https://github.com/actions/runner-images/tree/main/images/macos

Reply:
I am also confused, but since macOS 14 is designed for arm64 architecture, shouldn't see an x86 runner

Reply:
Supported by both x86 and arm64. 
There are still plenty of Mac having an x86 CPUs